xarray/backends/api.py

```diff
@@ -383,6 +384,10 @@ def open_dataset(
         represented using ``np.datetime64[ns]`` objects.  If False, always
         decode times to ``np.datetime64[ns]`` objects; if this is not possible
         raise an error.
+    overwrite_encoded_chunks: bool, optional
+        Whether to drop the zarr chunks encoded for each variable when a
+        dataset is loaded with specified chunk sizes (default: False)
```

Comment:
Since this only applies to the zarr backend, would it make sense to pass this through `backend_kwargs`?

Reply:
You are right, it does make sense to not make it an argument for ``open_dataset``, and pass it to ``open_dataset`` through ``backend_kwargs`` from depricated ``open_zarr`` (i.e. zarr.py). However, it is not an argument for ``ZarrStore.open_group``, instead it is used within ``def maybe_chunk``. I had to pop-it and use it before passing ``backend_kwargs`` to ``open_group`` on zarr.py

Reply:
Could we add `overwrite_encoded_chunks` as a new argument in `ZarrStore.open_group`? That seems preferable to adding another highly backend specific method on to `open_dataset`.

(To be honest, I'm not entirely sure why we need `overwrite_encoded_chunks`... it's not obvious from the description why it's useful)

Reply:
Quoting from https://github.com/pydata/xarray/pull/2530#issue-227407536, where `overwrite_encoded_chunks` was initially added.

>  It seems reasonable that anyone manually specifying chunks may want to rewrite the dataset in those chunks, and the error that arises when the encoded Zarr chunks mismatch the variable Dask chunks may quickly get annoying. `overwrite_encoded_chunks=True` sets the encoded chunks to None so there is no clash.

As mentioned by @Mikejmnez , `overwrite_encoded_chunks` is not actually used by `ZarrStore.open_group`, it is used at the `open_dataset` level. So it's not strictly a 'backend_kwarg', but we can still put it in there I suppose? Just looking through the code now to see how best to do it.